gpflow/config/__config__.py

```diff
@@ -124,12 +139,17 @@ def set_default_jitter(value: float):
     set_config(replace(config(), jitter=value))
 
 
-def set_default_summary_fmt(value: str):
-    formats = tabulate.tabulate_formats + ['notebook', None]
-    if value not in formats:
-        raise ValueError(f"Summary does not support '{value}' format")
+def set_default_positive_bijector(value: str):
+    if not isinstance(value, str):
+        raise TypeError(f"Expected a string, but got a {type(value)}")
 
-    set_config(replace(config(), summary_fmt=value))
+    valid_bijectors = {"exp", "softplus"}
```

Comment:
Could this simply be taken as `gpflow.utilities.bijectors._POSITIVE_BIJECTOR_MAP.keys()` instead? More DRY.

Reply:
@st-- , @jahall 

Yes! But, it should be in the config itself as an immutable field. The other way around is wrong.

```python
@dataclass(frozen=True)
class Config:
    """
    Immutable object for storing global GPflow settings
    Args:
        int: Integer data type, int32 or int64.
        float: Float data type, float32 or float64
        jitter: Jitter value. Mainly used for for making badly conditioned matrices more stable.
            Default value is `1e-6`.
        positive_bijector: Method for positive bijector, either "softplus" or "exp".
            Default is "softplus".
        positive_minimum: Lower level for the positive transformation.
        summary_fmt: Summary format for module printing.
    """

    int: type = field(default_factory=lambda: default(_Values.INT))
    float: type = field(default_factory=lambda: default(_Values.FLOAT))
    jitter: float = field(default_factory=lambda: default(_Values.JITTER))
    positive_bijector: str = field(default_factory=lambda: default(_Values.POSITIVE_BIJECTOR))
    positive_minimum: float = field(default_factory=lambda: default(_Values.POSITIVE_MINIMUM))
    summary_fmt: str = field(default_factory=lambda: default(_Values.SUMMARY_FMT))
    positive_type: Dict[str, type] = dict(exp=tfp.bijectors.Exp, softplus=tfp.bijectors.Softplus)
```

The only thing I don't like is that dictionary itself here is immutable.

Reply:
Yep, agree that config shouldn't be importing from other modules. Happy to implement @awav's suggestion.

Reply:
I refactored so that the actual tfp.bijector.Bijector instance lives in `Config.positive_bijector` rather than a plain string so should resolve these issues...though maybe we don't want "complex" datastructures like these in the config?